cmd/podman/pod_restart.go

```diff
@@ -42,37 +42,60 @@ func init() {
 }
 
 func podRestartCmd(c *cliconfig.PodRestartValues) error {
-	runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+	var lastError error
+	runtime, err := adapter.GetRuntime(&c.PodmanCommand)
 	if err != nil {
 		return errors.Wrapf(err, "could not get runtime")
 	}
 	defer runtime.Shutdown(false)
 
-	// getPodsFromContext returns an error when a requested pod
-	// isn't found. The only fatal error scenerio is when there are no pods
-	// in which case the following loop will be skipped.
-	pods, lastError := getPodsFromContext(&c.PodmanCommand, runtime)
+	restartIDs, conErrors, restartErrors := runtime.RestartPods(getContext(), c)
 
-	ctx := getContext()
-	for _, pod := range pods {
-		ctr_errs, err := pod.Restart(ctx)
-		if ctr_errs != nil {
-			for ctr, err := range ctr_errs {
-				if lastError != nil {
-					logrus.Errorf("%q", lastError)
-				}
-				lastError = errors.Wrapf(err, "unable to restart container %q on pod %q", ctr, pod.ID())
-			}
-			continue
-		}
-		if err != nil {
+	for _, p := range restartIDs {
+		fmt.Println(p)
+	}
+	if conErrors != nil && len(conErrors) > 0 {
+		for ctr, err := range conErrors {
 			if lastError != nil {
 				logrus.Errorf("%q", lastError)
 			}
-			lastError = errors.Wrapf(err, "unable to restart pod %q", pod.ID())
-			continue
+			lastError = errors.Wrapf(err, "unable to pause container %s", ctr)
 		}
-		fmt.Println(pod.ID())
+	}
+	if len(restartErrors) > 0 {
+		lastError = restartErrors[len(restartErrors)-1]
+		// Remove the last error from the error slice
+		restartErrors = restartErrors[:len(restartErrors)-1]
+	}
+	for _, err := range restartErrors {
+		logrus.Errorf("%q", err)
 	}
 	return lastError
+	// getPodsFromContext returns an error when a requested pod
```

Comment:
Why all these comments?

Reply:
just leftover from slicing and dicing, fixed.